src/Orleans.Core/Async/TaskExtensions.cs

```diff
@@ -198,7 +198,7 @@ internal static T WaitForResultWithThrow<T>(this Task<T> task, TimeSpan timeout)
         /// <param name="timeout">Amount of time to wait before timing out</param>
         /// <exception cref="TimeoutException">If we time out we will get this exception</exception>
         /// <returns>The completed task</returns>
-        internal static async Task WithTimeout(this Task taskToComplete, TimeSpan timeout)
+        public static async Task WithTimeout(this Task taskToComplete, TimeSpan timeout, string exceptionMessage = null)
```

Comment:
Does this need to be public?

Reply:
yeah. There's an identical one in Orleans.TestingHost. to make function less ambiguous, I made this public and removed that one. And `WithTimeout<T>` is already public, so I think it is fine to make this public,

Reply:
Ok, seems reasonable. Maybe they should both be internal, but at least this is more consistent now.